pkgs/tools/networking/curl/default.nix

```diff
@@ -177,16 +180,22 @@ stdenv.mkDerivation rec {
     ln $out/lib/libcurl${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-gnutls${stdenv.hostPlatform.extensions.sharedLibrary}.4.4.0
   '';
 
-  passthru = {
+  passthru = let
+    useThisCurl = attr: attr.override { curl = finalAttrs.finalPackage; };
+  in {
     inherit opensslSupport openssl;
     tests = {
-      inherit curlpp coeurl;
-      haskell-curl = haskellPackages.curl;
-      ocaml-curly = ocamlPackages.curly;
-      php-curl = phpExtensions.curl;
-      pycurl = python3.pkgs.pycurl;
+      curlWithTests = finalAttrs.finalPackage.overrideAttrs (_: { doCheck = true; });
+      curlpp = useThisCurl curlpp;
+      coeurl = useThisCurl coeurl;
```

Comment:
```nix
      fetchpatch = tests.fetchpatch.simple.override { fetchpatch = fetchpatch.override { fetchurl = useThisCurl fetchurl; }; };
```

where `tests` is `pkgs.tests`.

Haven't tried this, but would be pretty great to have something like this.
Also would be nice to have a direct fetchurl test. :shrug:

Reply:
I didn't actually want to repeat `.simple` here, but I don't think we have an `overridePackages` to complement `callPackages` yet. That could use `recurseForDerivations` to do the right thing.